qiskit/circuit/measure.py

```diff
@@ -41,6 +47,71 @@ def broadcast_arguments(self, qargs, cargs):
             raise CircuitError("register size error")
 
 
+class MeasureX(Measure):
+    """Quantum measurement in the X basis."""
+
+    basis: str | None = "X"
+
+    def __init__(self):
+        """Create new X measurement instruction."""
+        # pylint: disable=bad-super-call
+        super(Measure, self).__init__("measure_x", 1, 1, [])
+
+    def _define(self):
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+        qc = QuantumCircuit(1, 1)
+        qc.h(0)
+        qc.measure_z(0, 0)
+        qc.h(0)
+
+        self.definition = qc
+
+
+class MeasureY(Measure):
+    """Quantum measurement in the Y basis."""
+
+    basis: str | None = "Y"
+
+    def __init__(self):
+        """Create new Y measurement instruction."""
+        # pylint: disable=bad-super-call
+        super(Measure, self).__init__("measure_y", 1, 1, [])
+
+    def _define(self):
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+        qc = QuantumCircuit(1, 1)
+        qc.sdg(0)
+        qc.measure_x(0, 0)
+        qc.s(0)
+
+        self.definition = qc
+
+
+class MeasureZ(Measure):
+    """Quantum Z measurement in the Z basis."""
+
+    basis: str | None = "Z"
+
+    def __init__(self):
+        """Create new measurement instruction."""
+        # pylint: disable=bad-super-call
+        super(Measure, self).__init__("measure_z", 1, 1, [])
+
+    def _define(self):
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+        qc = QuantumCircuit(1, 1)
+        qc.measure(0, 0)
+
+        self.definition = qc
+
+
+# TODO: deprecated ? delete : update
 def measure(circuit, qubit, clbit):
```

Comment:
The docstring in this function seems to read that it has been deprecated since 0.19. Can someone confirm that this is safe to delete now?

Reply:
Yeah I think it's safe to remove it in a separate PR as it's been deprecated for ~1yr already.

Reply:
PR ready: #9278